src/mutation.cpp

```diff
@@ -896,30 +898,50 @@ bool Character::mutation_ok( const trait_id &mutation, bool force_good, bool for
     }
 
     const mutation_branch &mdata = mutation.obj();
-    if( force_bad && mdata.points > 0 ) {
-        // This is a good mutation, and we're due for a bad one.
+    if( !allow_good && mdata.points > 0 ) {
         return false;
     }
 
-    if( force_good && mdata.points < 0 ) {
-        // This is a bad mutation, and we're due for a good one.
+    if( !allow_bad && mdata.points < 0 ) {
+        return false;
+    }
+
+    if( !allow_neutral && mdata.points == 0 ) {
         return false;
     }
 
     return true;
 }
 
+bool Character::roll_bad_mutation() const
+{
+    //Instability value at which bad mutations become possible
+    const float I0 = 900.0;
+    //Instability value at which good and bad mutations are equally likely
+    const float I50 = 2800.0;
+
+    //A curve that is 0 until I0, crosses 0.5 at I50, then slowly approaches 1
+    float chance = std::max( 0.0, 1 - std::pow( I0 / vitamin_get( vitamin_instability ),
+                             std::log( 2 ) / std::log( I50 / I0 ) ) );
```

Comment:
`std::log` can't be constexpr but you can hardcode the result of `std::log( 2 ) / std::log( I50 / I0 ) )` instead and keep the formula in a comment.

Reply:
I did that, but on second thought, I think I can make it static instead. That way it doesn't have to be manually maintained but it will still only calculate once, right?